### DISCUSSION TITLE
Mvvm support for Snackbar

### DISCUSSION THREAD

**Discussion by Softtinn** [xamarin/XamarinCommunityToolkit#1266] [Q&A] [answered]
Hi!
I'm interested in using the XCT snackbar. However, from what I understand when examining the sample app, it's done entirely in the page code behind? Isn't there a built in MVVM support for it? If so; how should it be done? Any docs available?

**Comment by Softtinn** [comment, marked as answer]
I have a somewhat working solution now. I'm injecting the Prism EventAggregator in the codebehind with a custom event containing all Snackbar information I need. Still open for suggestions, but I think this'll do for now.

**Reply by developer9969** [reply]
Hi there I have the same issue. We need to show the snackbar/toast once a certain action has been completed and the idea of sending an event so that we can hook up on a page is exactly what I am looking for.

Do you mind sharing a snippet?  How do you subscribe to an event aggregator in a page. Thanks

**Reply by Softtinn** [reply]
Hi!

First off, we're using Prism (https://prismlibrary.com/) as our MVVM framework. Then we inject the Prism EventAggregator in the page constructor:

```
        private readonly IEventAggregator _eventAggregator;

        public ContentPage(IEventAggregator eventAggregator)
        {
            InitializeComponent();

            _eventAggregator = eventAggregator;
        }

        protected override void OnAppearing()
        {
            _eventAggregator?.GetEvent<MessageEvent>().Subscribe(DisplayMessage);
            base.OnAppearing();
        }

        protected override void OnDisappearing()
        {
            _eventAggregator?.GetEvent<MessageEvent>().Unsubscribe(DisplayMessage);
            base.OnDisappearing();
        }

        private async void DisplayMessage(SnackBarContent message)
        {
            var result = await this?.DisplaySnackBarAsync(message.Message, message.Action);
        }
```

We are using a custom extension for the Snackbar event which basically only takes the LogLevel severity in consideration.

Best of luck!

**Reply by developer9969** [reply]
Thank you for your reply. We use Prism too so this is exactly what I was looking for.

**Comment by maxkoshevoi** [comment]
From what I understand, MVVM is intended to sepparate UI from logic, so that that logic can be tested. Snackbar (same as Alert) is a UI, so it's call should be located in the page. View Model can send some event to the page that would trigger the Snackbar, or something.

Anyway, you can use `Shell.Current.CurrentPage` if you want to show Snackbar from the VM. But since 1.2.0 Snackbar can be displayed on top of an anchor element (floating action button for example), and at this point it's really not worth the effort to keep Snackbar in the VM.
